frontend/src/app/chatmenu/SearchMessages.stories.tsx

```diff
@@ -9,8 +10,34 @@ export default {
 
 const Template = (args) => <SearchMessages {...args} />
 
-// TODO: Set args with the props to test a case of your component
-export const Case1 = Template.bind({})
-Case1.args = {}
+const baseChatData = {
+    chat: {
+        for: 'asabe',
+        participants: {
+            asabe: getMockUser('asabe'),
+            pradeep: getMockUser('pradeep'),
+        },
+    },
+    messages: [],
+}
+
+const shortMessages = [
+    {
+        from: 'pradeep',
+        type: 'MESSAGE',
+        timestamp: 0,
+        message: "Hey, what's up?",
+    },
+    {
+        from: 'asabe',
+        type: 'MESSAGE',
+        timestamp: 1,
+        message: 'Not much, how about you?',
+    },
+]
+
+export const EmptySearch = Template.bind({})
+EmptySearch.args = { ...baseChatData }
 
-// TODO: Add more cases to test your component
+export const FoundSearch = Template.bind({})
+FoundSearch.args = { ...baseChatData, messages: [...shortMessages] }
```

Comment:
Nice job setting up these stories using the mock users and destructuring!

You don't have to do this, because this feature is working and the stories provide a way to test the search feature, but another way to make this code even more testable would be to separate the search bar and the filtered messages into their own components. Then you can pass in a search term to the filtered messages component to test whether it filters correctly. This allows you to decouple testing the filtering from the input, and test many edge cases quickly (lowercase, trimming, partial matches, etc.).

This is a common React pattern of splitting a feature into smaller components and "lifting state up" so that each component is not tightly coupled to another. If you are interested, you can read more about this in the official React docs: [Lifting State Up](https://reactjs.org/tutorial/tutorial.html#lifting-state-up).

Reply:
Great point here! @vingkan 
I'll set this up as an upcoming task and pair program together to separate out these components and add search string based test cases.